File: spar/src/main/java/com/project/spar/model/DeviceSpecification.java
package com.project.spar.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Data @NoArgsConstructor
@Entity @Table(name="device_specifications")
public class DeviceSpecification {
    @Id @GeneratedValue(strategy=GenerationType.IDENTITY) private Long id;
    @Column(nullable=false, unique=true) private String deviceId;
    private String deviceName;
    private String manufacturer;
    private String model;
    private String processor;
    private int cpuPhysicalCores;
    private int cpuLogicalCores;
    private double installedRam; // in GB
    private String graphics;
    private String operatingSystem;
    private String systemType;
    private LocalDateTime registeredAt = LocalDateTime.now();

    @ManyToOne(fetch=FetchType.LAZY)
    @JoinColumn(name="user_id", nullable=false)
    private User user;
}

File: SPAR_iOS/SPAR/SPAR/Data/DeviceSpecification.swift
//
//  DeviceSpecification.swift
//  SPAR
//
//  Created by Abhijeet Cherungottil on 4/19/25.
//

import Foundation

struct DeviceSpecification: Codable, Identifiable {
    let id: Int
    let deviceId: String
    let deviceName: String
    let manufacturer: String
    let model: String
    let processor: String
    let cpuPhysicalCores: Int
    let cpuLogicalCores: Int
    let installedRam: Double
    let graphics: String
    let operatingSystem: String
    let systemType: String
    let registeredAt: String
    
    enum CodingKeys: String, CodingKey {
        case id
        case deviceId,deviceName, manufacturer, model, processor
        case cpuPhysicalCores, cpuLogicalCores, installedRam
        case graphics, operatingSystem, systemType, registeredAt
    }
}
